CMake/AbseilFindDependencies.cmake

```diff
@@ -0,0 +1,121 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include(ExternalProject)
+
+
+#
+## pthread
+#
+find_package(Threads REQUIRED)
+
+#
+## CCTZ
+#
+if(NOT ABSL_ENABLE_BUNDLE)
+
+    find_package(CCTZ REQUIRED)
+
+else(NOT ABSL_ENABLE_BUNDLE)
+
+    find_package(CCTZ QUIET)
+    # embedded dep
+    if(NOT CCTZ_FOUND)
+        # if not found, try to bundle it
+        message(STATUS "No External CCTZ found, Use bundle CCTZ package")
+
+        set(CCTZ_BUNDLE_INTERNAL_INSTALLDIR "${CMAKE_CURRENT_BINARY_DIR}/CCTZ_BUNDLE-prefix/install")
+        set(CCTZ_BUNDLE_INTERNAL_SRCDIR "${CMAKE_CURRENT_BINARY_DIR}/CCTZ_BUNDLE-prefix/src/CCTZ_BUNDLE")
+
+        ExternalProject_Add(CCTZ_BUNDLE
+            GIT_REPOSITORY "https://github.com/google/cctz.git"
+            GIT_TAG "master"
+    #        SOURCE_DIR "${CCTZ_BUNDLE_INTERNAL_SRCDIR}"
+            CONFIGURE_COMMAND "true"                            #  pure makefile, no configure
+            BUILD_COMMAND "make" "PREFIX=${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}"
+                "SRC=${CCTZ_BUNDLE_INTERNAL_SRCDIR}/" "-f" "${CCTZ_BUNDLE_INTERNAL_SRCDIR}/Makefile"
+                "CXX=${CMAKE_CXX_COMPILER}" "CC=${CMAKE_C_COMPILER}"
+            INSTALL_DIR "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}"
+            INSTALL_COMMAND "make" "install"
+                "PREFIX=${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}" "SRC=${CCTZ_BUNDLE_INTERNAL_SRCDIR}/" "-f" "${CCTZ_BUNDLE_INTERNAL_SRCDIR}/Makefile"
+                "CXX=${CMAKE_CXX_COMPILER}" "CC=${CMAKE_C_COMPILER}"
+         )
+
+
+        list(APPEND ABSEIL_PRE_DEPENDENCIES CCTZ_BUNDLE)
+
+        set(CCTZ_INCLUDE_DIRS "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}/include")
+        set(CCTZ_LIBRARIES "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}/lib/libcctz.a")
+
+        install(FILES "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}/lib/libcctz.a" DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
+        install(DIRECTORY "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}/include/cctz" DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR})
+    endif(NOT CCTZ_FOUND)
+endif(NOT ABSL_ENABLE_BUNDLE)
+
+include_directories(${CCTZ_INCLUDE_DIRS})
+
+
+
+if(BUILD_TESTING)
+
+
+    if(NOT ABSL_ENABLE_BUNDLE)
+
+        find_package(GTest REQUIRED)
+        find_package(GMock REQUIRED)
+
+    else(NOT ABSL_ENABLE_BUNDLE)
+        # check for GTest
+        find_package(GTest QUIET)
+        find_package(GMock QUIET)
+
+        # embedded dep
+        if((NOT GTEST_FOUND) OR (NOT GMOCK_FOUND))
+            # if not found, try to bundle it
+            message(STATUS "No External GTest found, Use bundle GTest package")
+
+            set(GTEST_BUNDLE_INTERNAL_INSTALLDIR "${CMAKE_CURRENT_BINARY_DIR}/GTEST_BUNDLE-prefix/install")
+
+            ExternalProject_Add(GTEST_BUNDLE
+                GIT_REPOSITORY "https://github.com/google/googletest/"
```

Comment:
same comment, an archive might be better suited.

Reply:
Done